frontend/app/src/MetricsManager.ts

```diff
@@ -113,6 +122,42 @@ export class MetricsManager {
     this.appHash = appHash
   }
 
+  // Set metrics url if sent by the host_config, otherwise fallback to retrieving the default
+  public setMetricsConfig = (metricsUrl = ""): void => {
+    // Don't send metrics if explicitly set to "off"
+    if (metricsUrl === "off") {
+      return
+    } else if (metricsUrl) {
+      this.metricsUrl = metricsUrl
+    } else {
+      this.requestDefaultMetricsConfig()
+    }
+  }
+
+  // Fallback - Checks if cached in localStorage, otherwise fetches the config from a default URL
+  private async requestDefaultMetricsConfig(): Promise<any> {
+    const isLocalStoreAvailable = localStorageAvailable()
+
+    if (isLocalStoreAvailable) {
+      const cachedConfig = localStorage.getItem("stMetricsConfig")
+      if (cachedConfig) {
+        this.metricsUrl = cachedConfig
+        return
+      }
+    }
+
+    const response = await fetch(DEFAULT_METRICS_CONFIG)
```

Comment:
In case the endpoint gets very slow to respond for whatever reason, does this impact the startup time of Streamlit? We could potentially add a timeout (https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static) e.g. set to ~5 seconds, to prevent this scenario.